src/upm/iModuleStatus.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Author: Henry Bruce <henry.bruce@intel.com>
+ * Copyright (c) 2014 Intel Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#pragma once
+
+#include <stdexcept>
+
+namespace upm
+{
+/**
+ * @brief Interface for Module Status. Sensor and Actuactor Interfaces Derive from this Interface.
+ */
+
+#define UPM_THROW(msg) throw std::runtime_error(std::string(__FUNCTION__) + ": " + (msg))
+
+class IModuleStatus 
+{
+public:
+   /**
+    * Determines if sensor module is configured correctly
+    * e.g. if it uses i2c communication can it configure registers
+    * for correct operation.
+    *
+    * @return true if correctly configured, false it not
+    */
+   virtual bool isConfigured() = 0;
+
+   /**
+    * Returns name of module. This is the string in library name after libupm_
+
+    * @return name of module
+    */
+    virtual const char* getModuleName() = 0;
```

Comment:
I chose to return `const char*` as STL does not define an ABI so exporting objects such as `std::string` from a shared library can be dangerous. However the mraa C++ API exports `std::string` so I should probably do likewise to be consistent.